utils/file_backend_local.go

```diff
@@ -76,7 +76,7 @@ func writeFileLocally(fr io.Reader, path string) (int64, *model.AppError) {
 	defer fw.Close()
 	written, err := io.Copy(fw, fr)
 	if err != nil {
-		return written, model.NewAppError("WriteFile", "api.file.write_file_locally.writing.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return 0, model.NewAppError("WriteFile", "api.file.write_file_locally.writing.app_error", nil, err.Error(), http.StatusInternalServerError)
```

Comment:
I checked this into `master` always returning `written` even if an error occurs, since it's idiomatic to still return the number of bytes written even if an error prevented them all from being written.

Reply:
Yes, the codes for this place I considered are not rigorous enough